apis/v1alpha1/gateway_types.go

```diff
@@ -68,21 +68,29 @@ type GatewaySpec struct {
 	// logical endpoints that are bound on this Gateway's addresses.
 	// At least one Listener MUST be specified.
 	//
-	// Each Listener in this array must have a unique Port field,
-	// however a GatewayClass may collapse compatible Listener
-	// definitions into a single implementation-defined acceptor
-	// configuration even if their Port fields would otherwise conflict.
+	// In general, each Listener in this array must specify a unique Port
+	// value.  However, an implementation MAY group Listeners by Port and
+	// then collapse each group of Listeners into a single Listener if the
+	// implementation determines that the Listeners in the group are
+	// "compatible". An implementation MAY also group together and collapse
```

Comment:
The first sentence seems to contradict the rest of this.

Reply:
I was trying to express the general case (ports must be unique) and the exception (implementations may merge listeners with the same port), but I'll remove the first sentence if it doesn't add anything since it is confusing.